x-pack/test/detection_engine_api_integration/security_and_spaces/group10/risk_engine/telemetry_usage.ts

```diff
@@ -0,0 +1,137 @@
+
+  describe('Risk engine telemetry', async () => {
+    const { indexListOfDocuments } = dataGeneratorFactory({
+      es,
+      index: 'ecs_compliant',
+      log,
+    });
+    const kibanaServer = getService('kibanaServer');
+
+    before(async () => {
+      await esArchiver.load('x-pack/test/functional/es_archives/security_solution/ecs_compliant');
+    });
+
+    after(async () => {
+      await esArchiver.unload('x-pack/test/functional/es_archives/security_solution/ecs_compliant');
+    });
+
+    beforeEach(async () => {
+      await cleanRiskEngineConfig({ kibanaServer });
+      await deleteRiskEngineTask({ es, log });
+      await deleteRiskScoreIndecies({ log, es });
+      await deleteAllAlerts(supertest, log, es);
+      await deleteAllRules(supertest, log);
+    });
+
+    describe('Risk engine not enabled', () => {
+      it('should has empty riskEngineMetrics', async () => {
+        await retry.try(async () => {
+          const stats = await getRiskEngineStats(supertest, log);
+          const expected = {};
+          expect(stats).to.eql(expected);
+        });
+      });
+    });
+
+    describe('Risk engine enabled', () => {
+      let hostId: string;
+      let userId: string;
+
+      beforeEach(async () => {
+        hostId = uuidv4();
+        const hostEvent = buildDocument({ host: { name: 'host-1' } }, hostId);
+        await indexListOfDocuments(
+          Array(10)
+            .fill(hostEvent)
+            .map((event, index) => ({
+              ...event,
+              'host.name': `host-${index}`,
+            }))
+        );
+
+        userId = uuidv4();
+        const userEvent = buildDocument({ user: { name: 'user-1' } }, userId);
+        await indexListOfDocuments(
+          Array(10)
+            .fill(userEvent)
+            .map((event, index) => ({
+              ...event,
+              'user.name': `user-${index}`,
+            }))
+        );
+
+        await createAndSyncRuleAndAlerts({
+          query: `id: ${userId} or id: ${hostId}`,
+          alerts: 20,
+          riskScore: 40,
+        });
+
+        await riskEngineRoutes.init();
+      });
+
+      afterEach(async () => {
+        await cleanRiskEngineConfig({ kibanaServer });
+        await deleteRiskEngineTask({ es, log });
+        await deleteRiskScoreIndecies({ log, es });
+        await deleteAllAlerts(supertest, log, es);
+        await deleteAllRules(supertest, log);
+      });
+
+      it('should return riskEngineMetrics with expected values', async () => {
+        await waitForRiskScoresToBePresent({ es, log, scoreCount: 20 });
+        await retry.try(async () => {
+          const {
+            all_risk_scores_index_size: allRiskScoreIndexSize,
+            unique_risk_scores_index_size: uniqueRiskScoreIndexSize,
+            ...otherStats
+          } = await getRiskEngineStats(supertest, log);
+          const expected = {
+            unique_host_risk_score_total: 0,
```

Comment:
Are these actually supposed to be 0, or is it just that the `latest` transform hasn't run in the usual case, here?

Could the transform running in time for these assertions cause this to fail?

Reply:
Yes, it is because Transform not running yet, and there no scores.

I wasn't able to have situation when there will some values in tests

Reply:
> Yes, it is because Transform not running yet, and there no scores.
> I wasn't able to have situation when there will some values in tests

The transform _is_ running, it's just _unlikely_ to complete by the time this assertion is made. I think there's still a race condition here, but if you're confident that we won't hit it, 👍